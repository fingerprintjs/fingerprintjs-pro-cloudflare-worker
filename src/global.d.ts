declare module '*.ejs' {
  const value: <T>(args: T) => string
  export default value
}
