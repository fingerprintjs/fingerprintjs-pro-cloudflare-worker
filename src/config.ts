export const config = {
  fpcdn: '__FPCDN__',
  ingressApi: '__INGRESS_API__',
}
