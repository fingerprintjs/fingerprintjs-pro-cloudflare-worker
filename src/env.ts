export type WorkerEnv = {
  WORKER_PATH: string | null
  AGENT_SCRIPT_DOWNLOAD_PATH: string | null
  GET_RESULT_PATH: string | null
  PROXY_SECRET: string | null
}

const Defaults: WorkerEnv = {
  WORKER_PATH: 'cf-worker',
  AGENT_SCRIPT_DOWNLOAD_PATH: 'agent',
  GET_RESULT_PATH: 'getResult',
  PROXY_SECRET: null,
}

function getVarOrDefault(variable: keyof WorkerEnv, defaults: WorkerEnv): (env: WorkerEnv) => string | null {
  return function (env: WorkerEnv): string | null {
    return (env[variable] || defaults[variable]) as string | null
  }
}

function isVarSet(variable: keyof WorkerEnv): (env: WorkerEnv) => boolean {
  return function (env: WorkerEnv): boolean {
    return env[variable] != null
  }
}

export const workerPathVarName = 'WORKER_PATH'
export const getWorkerPathVar = getVarOrDefault(workerPathVarName, Defaults)
export const isWorkerPathSet = isVarSet(workerPathVarName)

export const agentScriptDownloadPathVarName = 'AGENT_SCRIPT_DOWNLOAD_PATH'
const getAgentPathVar = getVarOrDefault(agentScriptDownloadPathVarName, Defaults)
export const isScriptDownloadPathSet = isVarSet(agentScriptDownloadPathVarName)

export function getScriptDownloadPath(env: WorkerEnv): string {
  const agentPathVar = getAgentPathVar(env)
  return `/${getWorkerPathVar(env)}/${agentPathVar}`
}

export const getResultPathVarName = 'GET_RESULT_PATH'
const getGetResultPathVar = getVarOrDefault(getResultPathVarName, Defaults)
export const isGetResultPathSet = isVarSet(getResultPathVarName)

export function getGetResultPath(env: WorkerEnv): string {
  const getResultPathVar = getGetResultPathVar(env)
  return `/${getWorkerPathVar(env)}/${getResultPathVar}`
}

export const proxySecretVarName = 'PROXY_SECRET'
const getProxySecretVar = getVarOrDefault(proxySecretVarName, Defaults)
export const isProxySecretSet = isVarSet(proxySecretVarName)

export function getProxySecret(env: WorkerEnv): string | null {
  return getProxySecretVar(env)
}

export function getStatusPagePath(env: WorkerEnv): string {
  return `/${getWorkerPathVar(env)}/status`
}
