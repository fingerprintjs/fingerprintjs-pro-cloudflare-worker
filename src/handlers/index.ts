import { handleDownloadScript } from './handleDownloadScript'
import { handleIngressAPI } from './handleIngressAPI'
import { handleStatusPage } from './handleStatusPage'

export { handleDownloadScript, handleIngressAPI, handleStatusPage }
