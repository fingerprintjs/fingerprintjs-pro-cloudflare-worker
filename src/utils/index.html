
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">98.29% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>288/293</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">97.56% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>80/82</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>59/59</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">98.2% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>273/278</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="addProxyIntegrationHeaders.ts"><a href="addProxyIntegrationHeaders.ts.html">addProxyIntegrationHeaders.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="addTrafficMonitoring.ts"><a href="addTrafficMonitoring.ts.html">addTrafficMonitoring.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="cookie.ts"><a href="cookie.ts.html">cookie.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="34" class="abs high">34/34</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="34" class="abs high">34/34</td>
	</tr>

<tr>
	<td class="file high" data-value="createErrorResponse.ts"><a href="createErrorResponse.ts.html">createErrorResponse.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="createResponseWithMaxAge.ts"><a href="createResponseWithMaxAge.ts.html">createResponseWithMaxAge.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="fetchCacheable.ts"><a href="fetchCacheable.ts.html">fetchCacheable.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="getCacheControlHeaderWithMaxAgeIfLower.ts"><a href="getCacheControlHeaderWithMaxAgeIfLower.ts.html">getCacheControlHeaderWithMaxAgeIfLower.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	</tr>

<tr>
	<td class="file high" data-value="getEffectiveTLDPlusOne.ts"><a href="getEffectiveTLDPlusOne.ts.html">getEffectiveTLDPlusOne.ts</a></td>
	<td data-value="96.29" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.29" class="pct high">96.29%</td>
	<td data-value="135" class="abs high">130/135</td>
	<td data-value="95.83" class="pct high">95.83%</td>
	<td data-value="48" class="abs high">46/48</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="96.21" class="pct high">96.21%</td>
	<td data-value="132" class="abs high">127/132</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="32" class="abs high">32/32</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="21" class="abs high">21/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="21" class="abs high">21/21</td>
	</tr>

<tr>
	<td class="file high" data-value="proxyEndpoint.ts"><a href="proxyEndpoint.ts.html">proxyEndpoint.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	</tr>

<tr>
	<td class="file high" data-value="returnHttpResponse.ts"><a href="returnHttpResponse.ts.html">returnHttpResponse.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="routing.ts"><a href="routing.ts.html">routing.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Mon Jul 03 2023 15:14:17 GMT+0000 (Coordinated Universal Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    