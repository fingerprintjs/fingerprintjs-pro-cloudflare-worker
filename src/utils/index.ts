import { getCacheControlHeaderWithMaxAgeIfLower } from './getCacheControlHeaderWithMaxAgeIfLower'
import { createErrorResponseForIngress, createErrorResponseForProCDN } from './createErrorResponse'
import { fetchCacheable } from './fetchCacheable'
import {
  addTrafficMonitoringSearchParamsForVisitorIdRequest,
  addTrafficMonitoringSearchParamsForProCDN,
} from './addTrafficMonitoring'
import { returnHttpResponse } from './returnHttpResponse'
import { addProxyIntegrationHeaders } from './addProxyIntegrationHeaders'
import { getEffectiveTLDPlusOne } from './getEffectiveTLDPlusOne'
import { Cookie, createCookieStringFromObject, createCookieObjectFromHeaderValue, filterCookies } from './cookie'
import { createRoute, addTrailingWildcard, removeTrailingSlashesAndMultiSlashes, replaceDot } from './routing'

export {
  createCookieStringFromObject,
  getCacheControlHeaderWithMaxAgeIfLower,
  addTrafficMonitoringSearchParamsForVisitorIdRequest,
  addTrafficMonitoringSearchParamsForProCDN,
  createCookieObjectFromHeaderValue,
  createErrorResponseForIngress,
  createErrorResponseForProCDN,
  addProxyIntegrationHeaders,
  addTrailingWildcard,
  removeTrailingSlashesAndMultiSlashes,
  replaceDot,
  getEffectiveTLDPlusOne,
  returnHttpResponse,
  filterCookies,
  fetchCacheable,
  createRoute,
  Cookie,
}
