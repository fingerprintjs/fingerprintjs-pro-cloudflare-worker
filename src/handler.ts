import { getScriptDownloadPath, getGetResultPath, getHealthCheckPath, WorkerEnv, getStatusPagePath } from './env'

import { createErrorResponseForIngress, createErrorResponseForProCDN } from './utils'
import { handleDownloadScript, handleIngressAPI, handleHealthCheck, handleStatusPage } from './handlers'
import { createRoute } from './utils'

export async function handleRequest(request: Request, env: WorkerEnv): Promise<Response> {
  const url = new URL(request.url)

  const scriptDownloadRoute = createRoute(getScriptDownloadPath(env))
  if (url.pathname.match(scriptDownloadRoute)) {
    try {
      return await handleDownloadScript(request)
    } catch (e) {
      return createErrorResponseForProCDN(e)
    }
  }

  const getResultRoute = createRoute(getGetResultPath(env))
  if (url.pathname.match(getResultRoute)) {
    try {
      return await handleIngressAPI(request, env)
    } catch (e) {
      return createErrorResponseForIngress(request, e)
    }
  }

  const healthRoute = createRoute(getHealthCheckPath(env))
  if (url.pathname.match(healthRoute)) {
    return handleHealthCheck(env)
  }

  const statusRoute = createRoute(getStatusPagePath(env))
  if (url.pathname.match(statusRoute)) {
    return handleStatusPage(env)
  }

  return new Response(JSON.stringify({ error: `unmatched path ${url.pathname}` }), { status: 404 })
}
