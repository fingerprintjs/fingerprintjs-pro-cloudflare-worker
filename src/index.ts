import { handleRequest } from './handler'
import { WorkerEnv } from './env'
import { returnHttpResponse } from './utils'

export default {
  async fetch(request: Request, env: WorkerEnv) {
    return handleRequest(request, env).then(returnHttpResponse)
  },
}
