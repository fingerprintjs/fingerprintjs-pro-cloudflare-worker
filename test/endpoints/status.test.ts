import worker from '../../src'
import { WorkerEnv } from '../../src/env'

describe('status page content', () => {
  beforeEach(() => {
    Object.defineProperty(globalThis, 'crypto', {
      value: {
        getRandomValues: () => new Uint8Array(24),
      },
    })
  })
  test('when all variables are set', async () => {
    const workerEnv: WorkerEnv = {
      WORKER_PATH: 'worker_path',
      PROXY_SECRET: 'proxy_secret',
      GET_RESULT_PATH: 'get_result',
      AGENT_SCRIPT_DOWNLOAD_PATH: 'agent_download',
    }
    const req = new Request('http://localhost/worker_path/status')
    const response = await worker.fetch(req, workerEnv)
    expect(await response.text()).toMatchSnapshot()
  })
  test('when proxy secret is not set', async () => {
    const workerEnv: WorkerEnv = {
      WORKER_PATH: 'worker_path',
      PROXY_SECRET: null,
      GET_RESULT_PATH: 'get_result',
      AGENT_SCRIPT_DOWNLOAD_PATH: 'agent_download',
    }
    const req = new Request('http://localhost/worker_path/status')
    const response = await worker.fetch(req, workerEnv)
    expect(await response.text()).toMatchSnapshot()
  })
  test('when get result path is not set', async () => {
    const workerEnv: WorkerEnv = {
      WORKER_PATH: 'worker_path',
      PROXY_SECRET: 'proxy_secret',
      GET_RESULT_PATH: null,
      AGENT_SCRIPT_DOWNLOAD_PATH: 'agent_download',
    }
    const req = new Request('http://localhost/worker_path/status')
    const response = await worker.fetch(req, workerEnv)
    expect(await response.text()).toMatchSnapshot()
  })
  test('when agent script download path is not set', async () => {
    const workerEnv: WorkerEnv = {
      WORKER_PATH: 'worker_path',
      PROXY_SECRET: 'proxy_secret',
      GET_RESULT_PATH: 'get_result',
      AGENT_SCRIPT_DOWNLOAD_PATH: null,
    }
    const req = new Request('http://localhost/worker_path/status')
    const response = await worker.fetch(req, workerEnv)
    expect(await response.text()).toMatchSnapshot()
  })
  test('when agent script download path and proxy secret are not set', async () => {
    const workerEnv: WorkerEnv = {
      WORKER_PATH: 'worker_path',
      PROXY_SECRET: null,
      GET_RESULT_PATH: 'get_result',
      AGENT_SCRIPT_DOWNLOAD_PATH: null,
    }
    const req = new Request('http://localhost/worker_path/status')
    const response = await worker.fetch(req, workerEnv)
    expect(await response.text()).toMatchSnapshot()
  })
})

describe('status page response headers', () => {
  test('CSP is set', async () => {
    const workerEnv: WorkerEnv = {
      WORKER_PATH: 'worker_path',
      PROXY_SECRET: 'proxy_secret',
      GET_RESULT_PATH: 'get_result',
      AGENT_SCRIPT_DOWNLOAD_PATH: 'agent_download',
    }
    const req = new Request('http://localhost/worker_path/status')
    const response = await worker.fetch(req, workerEnv)
    expect(response.headers.get('content-security-policy')).toMatch(
      /^default-src 'self'; img-src https:\/\/fingerprint\.com; style-src 'nonce-[\w=]+'$/,
    )
  })
})

describe('status page other HTTP methods than GET', () => {
  test('returns 405 when method is POST', async () => {
    const workerEnv: WorkerEnv = {
      WORKER_PATH: 'worker_path',
      PROXY_SECRET: 'proxy_secret',
      GET_RESULT_PATH: 'get_result',
      AGENT_SCRIPT_DOWNLOAD_PATH: 'agent_download',
    }
    const req = new Request('http://localhost/worker_path/status', { method: 'POST' })
    const response = await worker.fetch(req, workerEnv)
    expect(response.status).toBe(405)
  })
})
