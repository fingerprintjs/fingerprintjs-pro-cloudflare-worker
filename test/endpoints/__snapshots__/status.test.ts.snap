// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status page content when agent script download path and proxy secret are not set 1`] = `
"
  <html lang='en-US'>
  <head>
    <meta charset='utf-8'/>
    <title>Fingerprint Cloudflare Worker</title>
    <link rel='icon' type='image/x-icon' href='https://fingerprint.com/img/favicon.ico'>
    <style nonce='QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFB'>
      span {
        display: block;
        padding-top: 1em;
        padding-bottom: 1em;
        text-align: center;
      }
    </style>
  </head>
  <body>
  <span>Your worker is deployed</span>
  <span>
  Worker version: __current_worker_version__
  </span>
  
    <span>
    The following environment variables are not defined. Please reach out our support team.
    </span>
    
      <span>
      AGENT_SCRIPT_DOWNLOAD_PATH is not set
      </span>
      
      <span>
      PROXY_SECRET is not set
      </span>
      
  <span>
  Please reach out our support via <a href='mailto:support@fingerprint.com'>support@fingerprint.com</a> if you have any issues
  </span>
    
  </body>
  </html>
  "
`;

exports[`status page content when agent script download path is not set 1`] = `
"
  <html lang='en-US'>
  <head>
    <meta charset='utf-8'/>
    <title>Fingerprint Cloudflare Worker</title>
    <link rel='icon' type='image/x-icon' href='https://fingerprint.com/img/favicon.ico'>
    <style nonce='QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFB'>
      span {
        display: block;
        padding-top: 1em;
        padding-bottom: 1em;
        text-align: center;
      }
    </style>
  </head>
  <body>
  <span>Your worker is deployed</span>
  <span>
  Worker version: __current_worker_version__
  </span>
  
    <span>
    The following environment variables are not defined. Please reach out our support team.
    </span>
    
      <span>
      AGENT_SCRIPT_DOWNLOAD_PATH is not set
      </span>
      
  <span>
  Please reach out our support via <a href='mailto:support@fingerprint.com'>support@fingerprint.com</a> if you have any issues
  </span>
    
  </body>
  </html>
  "
`;

exports[`status page content when all variables are set 1`] = `
"
  <html lang='en-US'>
  <head>
    <meta charset='utf-8'/>
    <title>Fingerprint Cloudflare Worker</title>
    <link rel='icon' type='image/x-icon' href='https://fingerprint.com/img/favicon.ico'>
    <style nonce='QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFB'>
      span {
        display: block;
        padding-top: 1em;
        padding-bottom: 1em;
        text-align: center;
      }
    </style>
  </head>
  <body>
  <span>Your worker is deployed</span>
  <span>
  Worker version: __current_worker_version__
  </span>
  
    <span>
    All environment variables are set
    </span>
    
  <span>
  Please reach out our support via <a href='mailto:support@fingerprint.com'>support@fingerprint.com</a> if you have any issues
  </span>
    
  </body>
  </html>
  "
`;

exports[`status page content when get result path is not set 1`] = `
"
  <html lang='en-US'>
  <head>
    <meta charset='utf-8'/>
    <title>Fingerprint Cloudflare Worker</title>
    <link rel='icon' type='image/x-icon' href='https://fingerprint.com/img/favicon.ico'>
    <style nonce='QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFB'>
      span {
        display: block;
        padding-top: 1em;
        padding-bottom: 1em;
        text-align: center;
      }
    </style>
  </head>
  <body>
  <span>Your worker is deployed</span>
  <span>
  Worker version: __current_worker_version__
  </span>
  
    <span>
    The following environment variables are not defined. Please reach out our support team.
    </span>
    
      <span>
      GET_RESULT_PATH is not set
      </span>
      
  <span>
  Please reach out our support via <a href='mailto:support@fingerprint.com'>support@fingerprint.com</a> if you have any issues
  </span>
    
  </body>
  </html>
  "
`;

exports[`status page content when proxy secret is not set 1`] = `
"
  <html lang='en-US'>
  <head>
    <meta charset='utf-8'/>
    <title>Fingerprint Cloudflare Worker</title>
    <link rel='icon' type='image/x-icon' href='https://fingerprint.com/img/favicon.ico'>
    <style nonce='QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFB'>
      span {
        display: block;
        padding-top: 1em;
        padding-bottom: 1em;
        text-align: center;
      }
    </style>
  </head>
  <body>
  <span>Your worker is deployed</span>
  <span>
  Worker version: __current_worker_version__
  </span>
  
    <span>
    The following environment variables are not defined. Please reach out our support team.
    </span>
    
      <span>
      PROXY_SECRET is not set
      </span>
      
  <span>
  Please reach out our support via <a href='mailto:support@fingerprint.com'>support@fingerprint.com</a> if you have any issues
  </span>
    
  </body>
  </html>
  "
`;
